tests/consumer/test_oauth2.py

```diff
@@ -127,6 +129,24 @@ def test_authorized_url():
         )
 
 
+def test_authorized_url_invalid_response():
+    app, _ = make_app(debug=True)
+    with app.test_client() as client:
+        # reset the session before the request
+        with client.session_transaction() as sess:
+            sess["test-service_oauth_state"] = "random-string"
+        # make the request
+        with pytest.raises(MissingCodeError) as missingError:
+            client.get(
+                "/login/test-service/authorized?error_code=1349048&error_message=IMUSEFUL",
+                base_url="https://a.b.c",
+            )
+        assert str(missingError.value) == ('(missing_code) The redirect request did not contain a token. '
+                                           'Instead I got: {"error_message": "IMUSEFUL", "error_code": "1349048"}')
+
+
+
+
```

Comment:
Another option would be to add diff-quality to your build #alwaysBeSelling

Set fail-under and you can see the nice little X 

Less polished by lighter weight!